packages/rest/src/rest.application.ts

```diff
@@ -244,7 +244,7 @@ export class RestApplication extends Application implements HttpServerLike {
    */
   redirect(
     fromPath: string,
-    toPathOrUrl: string,
+    toPathOrUrl: string | Function,
```

Comment:
Please use a more specific type instead of `Function`, for example:

```ts
toPathOrUrl: string | LocationFactory
```

where `LocationFactory` can be defined as follows:

```ts
export type LocationFactory = (ctx: RedirectContext) => string;

export interface RedirectContext {
  request: Request,
  // etc.
}
```

Reply:
On the second thought: let's not invent any new `RedirectContext` class/interface and use the existing RequestContext as defined here:

https://github.com/strongloop/loopback-next/blob/c8f1a71fc9ae7f2331f52fe11af43fab567ba4d1/packages/rest/src/request-context.ts#L15-L21

I feel the additional fields like `basePath`, etc. may be useful in other places besides redirection, therefore it make sense to define them directly on the request context.

Reply:
>  it make sense to define them directly on the request context.

See https://github.com/strongloop/loopback-next/pull/2547